paraglob.cpp

```diff
@@ -0,0 +1,111 @@
+#include "paraglob.h"
+
+Paraglob::Paraglob(const std::vector<std::string> &patterns): Paraglob() {
+  for (std::string pattern : patterns) {
+    this->add(pattern);
+  }
+  this->compile();
+}
+
+void Paraglob::add(std::string pattern) {
+  // Get the meta words
+  std::vector<std::string> m_words = get_meta_words(pattern);
+  // Put them away
+  if (m_words.size() == 0) {
+    this->single_wildcards.push_back(pattern);
+  }
+  for (std::string meta_word : m_words) {
```

Comment:
Matter of taste, but generally we tend to use more white space for Zeek code, including in particular some empty lines between logical blocks to separate them visually. Also, single-line blocks don't really need the curly braces.